src/sentry/event_manager.py

```diff
@@ -820,9 +824,16 @@ def _save_aggregate(self, event, hashes, release, **kwargs):
         all_hashes = self._find_hashes(project, hashes)
 
         try:
-            existing_group_id = six.next(h.group_id for h in all_hashes if h.group_id is not None)
+            existing_group_id, existing_tombstone_id = six.next(
```

Comment:
At this point, I'd just restructure this into:

```python
existing_group_id = None
for hash in all_hashes:
  if hash.group_id is not None:
    existing_group_id = hash.group_id
    break
  if hash.group_tombstone_id is not None:
    raise DiscardedHash(...)
```

or,

```python
def get_group_id(hash):
  if hash.group_tombstone_id is not None:
    raise DiscardedHash(...)
  return hash.group_id

existing_group_id = six.next(
  itertools.ifilter(
    None,
    itertools.imap(get_group_id, hashes),
  ),
  None,
)
```

… depending on your stylistic preference.

I don't think there was any reason this was required to be a generator, so I think most people's preference would be the first option.